aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -195,13 +198,42 @@ public Completable load() {
                             return;
                         }
                     }
-                    // Publish outbox status upon loading
-                    publishCurrentOutboxStatus();
                     emitter.onComplete();
                 },
                 emitter::onError
             );
-        })
+        }).andThen(Observable.fromIterable(mutationQueue.iterator())
```

Comment:
Can you explain the purpose of this section?

Reply:
in the previous section the first page of results is loaded and then pushed up. After the mutation queue is drained, if there was more than one page of pending mutations on disk then the next page needs to be loaded in to the queue.

The reason this is after the initial is to ensure that publishCurrentOutboxStatus only fires when there are no pending mutations in the queue and none left in the db.

the original design had a bug that would fire the outbox is empty after the first 100 left the queue but this prevents that situation.